WcaOnRails/app/models/registration.rb

```diff
@@ -25,11 +26,29 @@ class Registration < ActiveRecord::Base
   end
 
   def pending?
-    accepted_at.nil?
+    accepted_at.nil? && deleted_at.nil?
```

Comment:
You already wrote a `accepted?` and a `deleted?` method, how about using them here?

```
def pending?
  !accepted? && !deleted?
end
```

(btw, this code is convincing me we should eventually have an explicit status field and a separate table to track changes to that status field.)